xds/src/main/java/io/grpc/xds/XdsLoadReportStore.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright 2019 The gRPC Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc.xds;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.protobuf.Duration;
+import io.envoyproxy.envoy.api.v2.core.Locality;
+import io.envoyproxy.envoy.api.v2.endpoint.ClusterStats;
+import io.envoyproxy.envoy.api.v2.endpoint.UpstreamLocalityStats;
+import io.grpc.ClientStreamTracer;
+import io.grpc.ClientStreamTracer.StreamInfo;
+import io.grpc.LoadBalancer.PickResult;
+import io.grpc.Metadata;
+import io.grpc.xds.XdsClientLoadRecorder.ClientLoadCounter;
+import java.util.HashMap;
+import java.util.Map;
+import javax.annotation.concurrent.NotThreadSafe;
+
+/**
+ * An {@link XdsLoadReportStore} instance holds the client side load stats for a cluster. Methods
+ * should be called in a synchronized context.
+ */
+@NotThreadSafe
+class XdsLoadReportStore {
+
+  private final String clusterName;
+  private final Map<Locality, XdsClientLoadRecorder.ClientLoadCounter> localityLoadCounters;
```

Comment:
There should be a way to remove a locality from the map if it no longer has traffic, probably because xDS traffic director sent a new list of Localities that don't include it any more.

This may be done when generating a report. If all counters in the report are 0, the Locality may be removed. Doing this with concurrency in mind would require some tricks.

Reply:
I added a `discardClientLoadCounter` method, which can be called in locality update for removed localities. For adding the counter, I kept it as it is, done when trying to intercept the `PickResult` while the counter is missing.

Reply:
Nvm. I will add a `createClientLoadCounter` method to be called in locality update for newly assigned localities. Current approach might have a race between counter being added/discarded, although it does no harm.